src/app/dataValidator/helpers/dataLoader/asyncValidateFilesInDirectory.test.js

```diff
@@ -27,25 +32,27 @@ describe('readdirSync helper', () => {
     fileToValidateSpy.mockRestore();
   });
 
-  it('should call readScenario for every file in the /data directory', () => {
+  it('should call readScenario for every file in the /data directory', async () => {
     const readScenarioSpy = jest.spyOn(readScenario, 'readScenario');
 
-    expectMethodToBeCalledTimes(79, readScenarioSpy);
+    await expectMethodToBeCalledTimes(44, readScenarioSpy);
   });
 
-  it('should call fileToValidate for only the valid json file in the /data directory', () => {
+  it('should call fileToValidate for only the valid json file in the /data directory', async () => {
     fileToValidateSpy = jest.spyOn(readScenario, 'fileToValidate');
 
-    expectMethodToBeCalledTimes(60, fileToValidateSpy);
+    await expectMethodToBeCalledTimes(30, fileToValidateSpy);
   });
 
-  it('should call fileToValidate for only the files in /data/prod/news', () => {
+  it('should call fileToValidate for only the files in /data/prod/news', async () => {
     fileToValidateSpy = jest.spyOn(readScenario, 'fileToValidate');
 
-    expectMethodToBeCalledTimes(2, fileToValidateSpy, './data/prod/news');
+    await expectMethodToBeCalledTimes(2, fileToValidateSpy, './data/prod/news');
   });
 
   it('should return a promise', () => {
-    expect(readdirSync('./data') instanceof Promise).toEqual(true);
+    expect(asyncValidateFilesInDirectory('./data') instanceof Promise).toEqual(
+      true,
```

Comment:
Argh, such a shame this now has to be on multiple lines! Maybe if we renamed to `readdirAsync` it would fit onto one line?

Reply:
Looking at the method's responsibility I feel like it should clarify that it is actually running the validator code, but I do agree the naming is too long. Would `asyncValidateDir` be a sensible naming in your opinion?